sage-ios/SAGE/SAGE/CheckinView.swift

```diff
@@ -0,0 +1,187 @@
+        self.addShadowToView(self.startButton)
+        let checkIcon: FAKIcon = FAKIonIcons.androidDoneIconWithSize(checkSize)
+        checkIcon.setAttributes([NSForegroundColorAttributeName: UIColor.whiteColor()])
+        self.startButton.setImage(checkIcon.imageWithSize(CGSizeMake(checkSize,checkSize)), forState: .Normal)
+        self.addSubview(self.startButton)
+        
+        let closeSize: CGFloat = 24
+        self.endButton.backgroundColor = UIColor.lightRedColor
+        self.endButton.setSize(width: buttonSize, height: buttonSize)
+        self.endButton.layer.cornerRadius = buttonSize / 2
+        self.addShadowToView(self.endButton)
+        let closeIcon: FAKIcon = FAKIonIcons.androidCloseIconWithSize(closeSize)
+        closeIcon.setAttributes([NSForegroundColorAttributeName: UIColor.whiteColor()])
+        self.endButton.setImage(closeIcon.imageWithSize(CGSizeMake(checkSize,checkSize)), forState: .Normal)
+        self.addSubview(self.endButton)
+        
+        self.timerView.backgroundColor = UIColor.whiteColor()
+        self.timerView.setSize(width: timerSize, height: timerSize)
+        self.timerView.layer.cornerRadius = timerSize/2
+        self.addShadowToView(self.timerView)
+        self.addSubview(self.timerView)
+        
+        self.timerLabel.textAlignment = .Center
+        self.timerLabel.font = UIFont.systemFontOfSize(22.0)
+        self.timerLabel.textColor = UIColor.lightRedColor
+        self.timerView.addSubview(self.timerLabel)
+        
+        self.timerArc.fillColor = UIColor.lightRedColor.CGColor
+        self.timerView.layer.addSublayer(self.timerArc)
+        
+        self.timerBubble.backgroundColor = UIColor(white: 0.90, alpha: 1)
+        self.timerBubble.setSize(width: timerSize, height: timerSize)
+        self.timerBubble.layer.cornerRadius = timerSize/2
+        self.timerView.insertSubview(self.timerBubble, atIndex: 0)
+    }
+    
+    private func showView(showView: UIView, hide hideView: UIView?, duration: NSTimeInterval) {
+        showView.transform = CGAffineTransformMakeScale(0.05, 0.05)
+        showView.alpha = 1
+        self.bringSubviewToFront(showView)
+        
+        UIView.animateWithDuration(duration,
+            delay: 0,
+            usingSpringWithDamping: UIConstants.defaultSpringDampening,
+            initialSpringVelocity: UIConstants.defaultSpringVelocity,
+            options:.CurveEaseInOut,
+            animations: { () -> Void in
+                showView.transform = CGAffineTransformIdentity
+                hideView?.alpha = 0
+            }) { (complete: Bool) -> Void in }
+    }
+    
+    private func addShadowToView(view: UIView) {
+        view.layer.shadowOffset = CGSizeMake(0,1)
+        view.layer.shadowRadius = 1.5
+        view.layer.shadowOpacity = shadowOpacity
+    }
+    
+    private func createTimerArcWithPercentage(percentage: CGFloat) -> CGPath {
+        let pi: CGFloat = 3.1415926
```

Comment:
shouldn't pi already be a constant in the default math library or something? are you sure you have to define it